haunted-houses/src/test/java/cz/muni/fi/pa165/hauntedhouses/SpecterDaoTests.java

```diff
@@ -0,0 +1,319 @@
+package cz.muni.fi.pa165.hauntedhouses;
+
+import cz.muni.fi.pa165.hauntedhouses.dao.AbilityDao;
+import cz.muni.fi.pa165.hauntedhouses.dao.HouseDao;
+import cz.muni.fi.pa165.hauntedhouses.dao.SpecterDao;
+import cz.muni.fi.pa165.hauntedhouses.model.Ability;
+import cz.muni.fi.pa165.hauntedhouses.model.House;
+import cz.muni.fi.pa165.hauntedhouses.model.Specter;
+
+import org.testng.annotations.Test;
+import org.testng.annotations.BeforeMethod;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTransactionalTestNGSpringContextTests;
+import org.testng.Assert;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceException;
+import javax.transaction.Transactional;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author David Hofman
+ */
+@Transactional
+@SpringBootTest
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+public class SpecterDaoTests extends AbstractTransactionalTestNGSpringContextTests {
+
+    @Autowired
+    private SpecterDao specterDao;
+
+    @Autowired
+    private AbilityDao abilityDao;
+
+    @Autowired
+    private HouseDao houseDao;
+
+    @Autowired
+    private EntityManager em;
+
+    private Specter s1;
+    private Specter s2;
+
+    @BeforeMethod
+    public void setup() {
+
+        s1 = new Specter();
+        s1.setName("SpecterName1");
+        s1.setDescription("SpecterDescription1");
+        s1.setStartOfHaunting(new Date());
+        s1.setEndOfHaunting(new Date());
+
+        s2 = new Specter();
+        s2.setName("SpecterName2");
+        s2.setDescription("SpecterDescription2");
+        s2.setStartOfHaunting(new Date());
+        s2.setEndOfHaunting(new Date());
+    }
```

Comment:
Mohol by si tu použiť konkrétny Date? V testoch nechceš mať nedeterministické chovanie a new Date() vytvorí Date s takým dátumom a časom (presným na milisekundy), kedy si ho vytvoril. Takže pri každom spustení testov tam bude iná hodnota. V tvojich testoch to nezaváži, pretože som si nevšimol, že by porovnával voči konkrétnej hodnote, ale len voči tomu, čo sa tam počas setup() metódy uložilo, ale je to bad practice.

Reply:
Co vůbec mají ty dva atributy představovat? Nějaký interval během dne, kdy má probíhat strašení nebo něco jiného? Navíc třída House používá pro podobné účely LocalDate, nechceme to tady taky sjednotit? Jen se ptám, abych to pak nemusel přepisovat vícekrát.

Reply:
Trieda Date je staršia a nebola úplne dobre navrhnutá, preto vznikol LocalDate. Mali by sme používať LocalDate.

Reply:
Date v sebe obsahuje dátum a aj čas. V Time API (java.time.*) to rozdelili na LocalDate a LocalTime, takže podľa mňa v Specter by sme mali použiť LocalTime a v House LocalDate

Reply:
Ok, dává to smysl. Potom testy opravím na LocalTime. Použití LocalTime tedy implikuje, že ty atributy představují interval během dne, kdy specter aktivně straší?

Reply:
Ano, podla zadania to je 1 interval behom dna.